sweep/sweeper.go

```diff
@@ -1558,3 +1569,166 @@ func (s *UtxoSweeper) sweepPendingInputs(inputs pendingInputs) {
 		}
 	}
 }
+
+// monitorFeeBumpResult subscribes to the passed result chan to listen for
+// future updates about the sweeping tx.
+//
+// NOTE: must run as a goroutine.
+func (s *UtxoSweeper) monitorFeeBumpResult(resultChan <-chan *BumpResult) {
+	defer s.wg.Done()
+
+	for {
+		select {
+		case r := <-resultChan:
+			// Validate the result is valid.
+			if err := r.Validate(); err != nil {
+				log.Errorf("Received invalid result: %v", err)
+				continue
+			}
```

Comment:
I don't know, just as I don't know when `NewBudgetInputSet` will have inputs that have different deadline heights, but we still validate because the data is provided by other subsystems. Same goes here - the result is given by the fee bumper, and the sweeper needs to make sure it gets what it expected.